client/CPlayerInterface.cpp

```diff
@@ -193,6 +193,8 @@ void CPlayerInterface::playerStartsTurn(PlayerColor player)
 
 void CPlayerInterface::performAutosave()
 {
+	std::string id = cb->getStartInfo()->uuid.substr(0, 8);

```

Comment:
if you need short uuid to 8 bytes, better to use CRC32, rather than substring to reduce duplication probability. Boost has implementation https://www.boost.org/doc/libs/1_83_0/doc/html/crc.html

Reply:
Yes. But I don't think that here is a real risk of collision?